pkg/server/server.go

```diff
@@ -712,6 +712,21 @@ func (s *Server) Write(ctx context.Context, req *openfgav1.WriteRequest) (*openf
 	})
 }
 
+// isDeadlineExceeded returns whether err is DeadlineExceeded or it is an internal error and the internal error
+// is DeadlineExceeded
+func isDeadlineExceeded(err error) bool {
```

Comment:
Can we colocate this alongside the errors code it is associated with in `pkg/server/errors`?

Reply:
Based on comment below, we don't need separate function to handle this and can be tested inline